ambry-tools/src/main/java/com.github.ambry/clustermap/HelixBootstrapUpgradeTool.java

```diff
@@ -0,0 +1,667 @@
+  private static final String SEALED_STR = "SEALED";
+  private static final String ZKINFO_STR = "zkInfo";
+  private static final String ZKCONNECTSTR_STR = "zkConnectStr";
+
+  static final int DEFAULT_MAX_PARTITIONS_PER_RESOURCE = 100;
+
+  /**
+   * @param args takes in three mandatory arguments: the hardware layout path, the partition layout path and the zk
+   *             layout path.
+   *             The Zk layout has to be of the following form:
+   *             {
+   *               "zkInfo" : [
+   *                 {
+   *                   "datacenter":"dc1",
+   *                   "zkConnectStr":"abc.example.com:2199",
+   *                 },
+   *                 {
+   *                   "datacenter":"dc2",
+   *                   "zkConnectStr":"def.example.com:2300",
+   *                 }
+   *               ]
+   *             }
+   *
+   *             Also takes in an optional argument that specifies the local datacenter name, so that can be used as
+   *             the "reference" datacenter. If none provided, the tool simply chooses one of the datacenters in the
+   *             layout as the reference datacenter.
+   */
+  public static void main(String args[]) throws Exception {
+    OptionParser parser = new OptionParser();
+
+    ArgumentAcceptingOptionSpec<String> hardwareLayoutPathOpt =
+        parser.accepts("hardwareLayoutPath", "The path to the hardware layout json file")
+            .withRequiredArg()
+            .describedAs("hardware_layout_path")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> partitionLayoutPathOpt =
+        parser.accepts("partitionLayoutPath", "The path to the partition layout json file")
+            .withRequiredArg()
+            .describedAs("partition_layout_path")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> zkLayoutPathOpt = parser.accepts("zkLayoutPath",
+        "The path to the json file containing zookeeper connect info. This should be of the following form: \n{\n"
+            + "  \"zkInfo\" : [\n" + "     {\n" + "       \"datacenter\":\"dc1\",\n"
+            + "       \"zkConnectStr\":\"abc.example.com:2199\",\n" + "     },\n" + "     {\n"
+            + "       \"datacenter\":\"dc2\",\n" + "       \"zkConnectStr\":\"def.example.com:2300\",\n" + "     },\n"
+            + "     {\n" + "       \"datacenter\":\"dc3\",\n" + "       \"zkConnectStr\":\"ghi.example.com:2400\",\n"
+            + "     }\n" + "  ]\n" + "}").
+        withRequiredArg().
+        describedAs("zk_connect_info_path").
+        ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> localDcOpt =
+        parser.accepts("localDc", "(Optional argument) The local datacenter name")
+            .withRequiredArg()
+            .describedAs("local_dc")
+            .ofType(String.class);
+
+    ArgumentAcceptingOptionSpec<String> maxPartitionsInOneResourceOpt = parser.accepts("maxPartitionsInOneResource",
```

Comment:
Do we have any suggestion as to how to set this no? During different upgrades, the total partitions added might be different right. Do you think adding a line to documentation is required?